src/Compilers/CSharp/Portable/Symbols/UsedAssembliesRecorder.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class UsedAssembliesRecorder : BoundTreeWalkerWithStackGuard
```

Comment:
What about:
* typeof
* default
* events
* exception variable declarations

Reply:
> What about:

I believe PR description clearly says what is in scope for this PR. BTW, ```typeof``` and "exception variable declarations" do not need special handling because the types are referred to explicitly in syntax and will be intercepted by ```Binder.ResultSymbol```.

---
In reply to: [347628028](https://github.com/dotnet/roslyn/pull/39807#discussion_r347628028) [](ancestors = 347628028)